model/src/main/kotlin/SnippetFinding.kt

```diff
@@ -26,9 +26,9 @@ package org.ossreviewtoolkit.model
  */
 data class SnippetFinding(
     /**
-     * The text location in the scanned source file where the snippet has matched.
+     * The text locations in the scanned source file where the snippet has matched.
      */
-    val sourceLocation: TextLocation,
+    val sourceLocation: TextWithMultipleLocations,
```

Comment:
Sorry for the (maybe naive) question, but why's this not simply a list / set of `TextLocation`?

Reply:
My assumption was that the reason for introducing the new class was to deduplicate the path.

Reply:
Actually, looking at the change again, I'm surprised that `val sourceLocation: TextLocation` is touched at all; I though what we discussed was to allow multiple snippets to be referenced from a single code location, so it's `val snippet: Snippet` that should change and become a collection, no?

Reply:
> This change of the first commit is something additional that needs to be done for compatibility with FossID.

Maybe I'm missing or mis-remembering something, but I thought we'd treat FossID snippet matches with multiple non-consecutive ranges as multiple individual `SnippetFinding`s in ORT's data model.

After all, we should create ORT's snippet data model after what we believe makes sense and what's able to cover all use-cases, and then map FossID's model to ORT's, instead of making ORT's model match FossID's exactly in all cases.

Reply:
I remember discussing that, if a snippet has multiple purls, it should be treated as multiple `SnippetFinding`. This is already the case in the implementation.
For non-consecutive line ranges, this is a good point and it could be that we agreed on the same and it felt under the radar.

However I worry this will again make the ORT result grow quite a bit. And I am not sure of the snippet choice could work with such split model.

I will put this PR back in draft and ponder this change. I also reached out to the ScanOSS team to see how they deal will non-consecutive line ranges as this could be interesting in designing the ORT model.

> map FossID's model to ORT's, instead of making ORT's model match FossID's exactly in all cases.

Yes of course and this is why I am always doing the splits when developing for Bosch and integrating in ORT 😄

Reply:
@sschuberth I received a response from the ScanOSS team.

> I'd like to clarify that the second option you mentioned is indeed the correct one. In this case, the engine will generate a unified string containing all the line ranges, each separated by a comma. Here's an example for better understanding:
> `"lines": "0-15, 20-32, 33-50, 65-83, 124-131, 161-172, 183-192, 204-221, 236-245, 247-254, 256-266"`

Therefore FossID and ScanOSS have the same kind of results with non consecutive lines. However, after experimenting, I think splitting non consecutive lines across multiple `SnippetFinding`s is ok for ORT. It is not the most compact model but I don't see any drawback in using it.

As a side note, an interesting other comment from ScanOSS:
 
> Furthermore, I wanted to bring your attention to the availability of a feature called "High Precision Snippet Matching" (HPSM). This feature can be utilized to attain even more accurate line ranges. Additionally, by using the "-F2" flag, you can incorporate snippet IDs into the results. I want to emphasize that both of these features are accessible through our clients.

So snippets in ScanOSS can have IDs as FossID does. Maybe we should, in a future PR, add the ID to the snippet model in ORT. Hence, we wouldn't have to store the ids in the snippet's `additionalData` as we currently do for FossID.
What do you think ?

Reply:
If you copy one snipped into multiple files to different line locations, do these get detected with identical snipped ID ?

Reply:
@fviernau For ScanOSS I cannot answer as I didn't test this `-F2` parameter.
For FossID, I would say 'no': If you scan a project and rescan it, the snippets are the same but with different ids.

Therefore, if we add this `id` property to the snippet model in ORT, I would put a documentation saying that its meaning is vendor specific.

Reply:
> For FossID, I would say 'no': If you scan a project and rescan it, the snippets are the same but with different ids.

Thanks, that's a pitty.